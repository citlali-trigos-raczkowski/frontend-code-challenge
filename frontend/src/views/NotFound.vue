<template>
  <div class="notFound">Oh no! Page was not found</div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "NotFound",
});
</script>
