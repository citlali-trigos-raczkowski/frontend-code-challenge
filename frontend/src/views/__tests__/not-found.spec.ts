import { shallowMount, createLocalVue } from "@vue/test-utils";
import NotFound from "../not-found.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("views/not-found.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(NotFound, {
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("includes link to Pikachu", async () => {
    const wrapper = shallowMount(NotFound, {
      localVue,
    });

    const aWrapper = wrapper.find("#not-found-link-pikachu");
    expect(aWrapper.attributes().to).toBe("/pikachu");
  });
  it("includes link home", async () => {
    const wrapper = shallowMount(NotFound, {
      localVue,
    });

    const aWrapper = wrapper.find("#not-found-link-home");
    expect(aWrapper.attributes().to).toBe("/");
  });
});
