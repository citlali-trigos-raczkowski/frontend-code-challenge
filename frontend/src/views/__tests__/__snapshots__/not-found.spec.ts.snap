// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/not-found.vue renders correctly 1`] = `
<div
  id="not-found-page"
>
  <router-link-stub
    ariacurrentvalue="page"
    data-cy="view-pikachu-link"
    event="click"
    id="not-found-link-pikachu"
    tag="a"
    to="/pikachu"
  >
    <img
      alt="pokemon-image"
      id="not-found-image"
      src="../../public/happy-pikachu.png"
    />
  </router-link-stub>
   
  <div
    id="not-found-info"
  >
    <h2>
      Oh no! That page was not found.
    </h2>
     
    <h3>
      Here's Pikachu instead.
    </h3>
     
    <br />
     
    <router-link-stub
      ariacurrentvalue="page"
      data-cy="return-home-link"
      event="click"
      id="not-found-link-home"
      tag="a"
      to="/"
    >
      
      Return home
    
    </router-link-stub>
     
    <h5>
      (or click Pikachu)
    </h5>
  </div>
</div>
`;
